casepro/profiles/signals.py

```diff
@@ -0,0 +1,20 @@
+from __future__ import unicode_literals
+
+from django.contrib.auth.models import User
+from django.db.models.signals import m2m_changed
+from django.dispatch import receiver
+
+from casepro.msgs.models import Label, Message
+
+from .models import Notification
+
+
+@receiver(m2m_changed, sender=Message.labels.through)
+def notify_label_watchers(sender, instance, action, reverse, model, pk_set, **kwargs):
```

Comment:
Signalophobe. The signal approach was useful at first because `post_add` tells you which labels were actually added at the database level (and aren't duplicates), but now the `Notification` model itself prevents duplicate notifications. Have added methods on `Message` for labelling/unlabelling and notifications are now created explicitly from there.